server/server.go

```diff
@@ -54,13 +69,16 @@ func NewServer(options *NewServerOptions) *Server {
 	router := mux.NewRouter()
 
 	srv := &Server{
-		router:         router,
-		store:          st,
-		pubSub:         rpc.NewPubSub(),
-		allowedOrigins: options.AllowedOrigins,
-		webhook:        options.Webhook,
-		serverURL:      options.ServerURL,
-		readDeadline:   options.ReadDeadline,
+		router:            router,
+		store:             st,
+		pubSub:            rpc.NewPubSub(),
```

Comment:
I'm not seeing the code that fixes the panic in `pub_sub.go` on line 142. With the current setup, WalletLink will crash if there's a race condition and we write to a close channel

<img width="507" alt="image" src="https://user-images.githubusercontent.com/41082194/82095580-28bd5d80-96b4-11ea-95dc-60297da9b526.png">

Reply:
I haven't seen any panics in `pub_sub.go` after this fix https://github.com/walletlink/walletlink/pull/29

The issue right now is the lock https://github.com/walletlink/walletlink/blob/master/server/rpc/pub_sub.go#L140 never get unlocked.

Reply:
We probably haven't seen it because of the lock issue you mentioned but once the lock issue is resolved this can crash the server i.e. we will be writing to a closed channel.

Can we fix it as part of this PR?

Reply:
As far as I know there is no way to check if a channel is closed in go. Do you any suggestions to prevent the app from crashing?